clients/src/main/java/io/strimzi/http/consumer/HttpConsumerClient.java

```diff
@@ -156,7 +157,8 @@ public void subscribeToTopic() {
     }
 
     public void consumeMessages() {
-        HttpContext context = HttpContext.get(configuration.getConsumeMessagesURI(), ConfigurationConstants.HTTP_JSON_CONTENT_TYPE);
+        String contentType = "application/vnd.kafka." + this.configuration.getMessageType() + ".v2+json";
```

Comment:
Would it make sense to have the content type as part of the HTTP configuration class, so you don't need to build it on more places?